sync/src/main/java/tech/pegasys/artemis/sync/PeerSync.java

```diff
@@ -48,12 +48,20 @@ public PeerSync(final ChainStorageClient storageClient, final BlockImporter bloc
   }
 
   public SafeFuture<PeerSyncResult> sync(final Eth2Peer peer) {
+    LOG.debug("Start syncing to peer {}", peer);
     // Begin requesting blocks at our first non-finalized slot
     final UnsignedLong finalizedEpoch = storageClient.getFinalizedEpoch();
     final UnsignedLong latestFinalizedSlot = compute_start_slot_at_epoch(finalizedEpoch);
     final UnsignedLong firstNonFinalSlot = latestFinalizedSlot.plus(UnsignedLong.ONE);
 
-    return executeSync(peer, peer.getStatus(), firstNonFinalSlot);
+    return executeSync(peer, peer.getStatus(), firstNonFinalSlot)
+      .whenComplete((res,err) -> {
+        if (err != null) {
+          LOG.debug("Failed to sync with peer {}: {}", peer, err);
```

Comment:
Note that this will print the exception message, but not the full stack trace. Not sure if that's what you wanted or not.  You can't use placeholders and get a full stack trace.